src/climate_data/data.py

```diff
@@ -982,6 +982,269 @@ def load_results(
         return pd.read_parquet(path)
 
 
+class FloodingData:
+    """Class for managing the flooding data used in the project."""
+
+    def __init__(
+        self,
+        root: str | Path = cdc.FLOOD_ROOT,
+        *,
+        read_only: bool = False,
+    ) -> None:
+        self._root = Path(root)
+        self._credentials_root = self._root / "credentials"
+        self._read_only = read_only
+        if not read_only:
```

Comment:
You don't want this here.  The flooding directory is _always_ read-only in this repository.  I have it for the `ClimateData` class because this repo has pipeline stages where climate data is an output directory and other stages that only consume from the climate data and write to the aggregates directory.